src/Teleportation.tsx

```diff
@@ -0,0 +1,167 @@
+import * as React from 'react'
+import { useThree } from '@react-three/fiber'
+import { Interactive, useXR } from '@react-three/xr'
+import { useCallback, useEffect, useState } from 'react'
+import { Quaternion, Vector3 } from 'three'
+
+export function useTeleportation() {
+  const [baseReferenceSpace, setBaseReferenceSpace] = useState<XRReferenceSpace | null>(null)
+  const gl = useThree((state) => state.gl)
+  const { session } = useXR()
+
+  useEffect(() => {
+    const b = gl.xr.getReferenceSpace()
+    if (b) {
+      setBaseReferenceSpace(b)
+    }
+    const onSessionStart = () => {
+      if (gl.xr) {
+        setBaseReferenceSpace(gl.xr.getReferenceSpace())
+      }
+    }
+    gl.xr.addEventListener('sessionstart', onSessionStart)
+    return () => {
+      gl.xr.removeEventListener('sessionstart', onSessionStart)
+    }
+  }, [gl])
+
+  const teleportTo = useCallback(
+    async (worldPosition: Vector3) => {
+      if (baseReferenceSpace && session) {
+        const pose = await new Promise<XRViewerPose | undefined>((resolve) => {
+          session.requestAnimationFrame((_, xrFrame) => {
+            const pose = xrFrame.getViewerPose(baseReferenceSpace)
+            resolve(pose)
+          })
+        })
+
+        const offsetX = pose?.transform.position.x || 0
+        const offsetZ = pose?.transform.position.z || 0
+
+        const offsetFromBase = {
+          x: -worldPosition.x + offsetX,
+          y: -worldPosition.y,
+          z: -worldPosition.z + offsetZ
+        }
+        const transform = new XRRigidTransform(offsetFromBase, new Quaternion())
+        const teleportSpaceOffset = baseReferenceSpace.getOffsetReferenceSpace(transform)
+        gl.xr.setReferenceSpace(teleportSpaceOffset)
+      }
+    },
+    [baseReferenceSpace]
+  )
+
+  return {
+    teleportTo
+  }
+}
+
+export type TeleportationPlaneProps = {
+  leftHand?: boolean
+  rightHand?: boolean
+  maxDistance?: number
+}
+
+const MARKER_SIZE = 0.25
+
+export function TeleportationPlane(props: TeleportationPlaneProps) {
+  const { teleportTo } = useTeleportation()
+  const [intersection, setIntersection] = useState<Vector3 | null>(null)
+  const { controllers } = useXR()
+  const [size, setSize] = useState(MARKER_SIZE)
+  const maxDistanceTeleport = props.maxDistance || 10
+  const { camera } = useThree()
+
+  return (
+    <>
+      {intersection && (
+        <mesh position={intersection} rotation={[-Math.PI / 2, 0, 0]}>
```

Comment:
Everything was working for me in my project with your code except for the white circle indicator. This seemed to fix it rendering

` <mesh position={[intersection?.x, 0, intersection?.z]} rotation={[-Math.PI / 2, 0, 0]}>`